### PR TITLE
Update Ember CLI to 4.2

### PR DESCRIPTION
**Pull request by jrjohnson**
With blueprint updates to get the latest ember version as well.

### CODE REVIEW

**Comment by jrjohnson** [comment]
Tests that are pending have been replaced and are no longer valid, I'll pull them out once this is merged.

**Review by stopfstedt — CHANGES_REQUESTED** [review]

File: tests/dummy/config/environment.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = function (environment) {
     modulePrefix: 'dummy',
     environment,
     rootURL: '/',
-    locationType: 'auto',
+    locationType: 'history',
```
[the remark this answers is not in the document]
Comment (jrjohnson) [reply tests/dummy/config/environment.js:16]: Auto is deprecated https://deprecations.emberjs.com/v4.x/#toc_deprecate-auto-location and history is the new ember-cli default.

**Review by stopfstedt — APPROVED** [review]
LGTM

**Review by dartajax — APPROVED** [review]
I click tested a bunch - looks fine. I'm assigned - gonna merge.

**Pull request merged**